### PR TITLE
Align Bloom Filter Hashing with Mainnet Compatible Clients

### PR DESCRIPTION
**Pull request by AusIV**
When clients like Go-ethereum and Parity create bloom filters, they trim leading zeroes off the topic before applying keccak to hash the value. As far as I can tell, this is inconsistent with the yellow paper (see: https://github.com/ethereum/yellowpaper/issues/690), but for ethereumjs-vm generated bloom filters to be compatible with tools that consume mainnet bloom filters, it should follow the same pattern.

See https://github.com/ethereum/go-ethereum/blob/master/core/types/bloom9.go#L60 where go-ethereum casts topics as big.Int then back to bytes, effectively stripping away leading zeroes before hashing.

### CODE REVIEW

**Comment by holgerd77** [comment]
Hi @AusIV, thanks for the PR!

I would generally support this. Any comments from other team members?

**Comment by holgerd77** [comment]
Can you rebase this (if you haven't done before you should do a local repository backup)? You also have some linting errors which prevent the tests above to pass (run ``npm run lint``), can you fix these? You can ease this by (temporarily) add ``--fix`` to the npm command in ``package.json``. Optimally you can then do a ``fixup`` with git rebase to have these changes included in the previous commit.

**Comment by AusIV** [comment]
I've made the requested changes.

**Comment by holgerd77** [comment]
Cool, thanks

**Comment by holgerd77** [comment]
Hi @AusIV, I would really like to get this merged. Could you maybe do a last rebase and rename to ``dropLeadingZeros`` like suggested? Would be cool if you find the time!

**Comment by holgerd77** [comment]
Hi @AusIV, thanks for taking up on this again, cool! 😄 

The ``add`` and the ``check`` functions still use the old name, this is why the tests are still failing.

I don't know if you hesitate on the rebase part if you haven't done rebasing that much before. Sorry for that, but we have relatively strict rules on that every PR needs to follow that
a) a PR has to be rebased on top of all other changes and
b) the commit history has to be somewhat clean so that other devs can follow up what is going on

If you want to prevent rebasing I would suggest the following:

0) Make a local backup of your repository
1) Delete your fork on GitHub and refork again, like this you have a clean and updated ``master`` branch
2) Create a new branch ``bloom-filter`` or something and manually add the changes from your backup repository
3) Do a new PR

I know this is a bit annoying but I think at the end that's the easiest way to go. I have also already gone through this and can say that it is actually less work than it sounds! 😄

**Comment by holgerd77** [comment]
Hi @AusIV, I would really like to get this merged. If you don't find any time to update this atm, would it be ok if I take your changes and submit a new PR on this?

File: lib/bloom.js [hunk 1]
```diff
@@ -41,7 +54,7 @@ Bloom.prototype.add = function (e) {
  * @param {Buffer} element
  */
 Bloom.prototype.check = function (e) {
-  e = utils.sha3(e)
+  e = utils.sha3(trimBuffer(e))
```
Comment (axic) [inline lib/bloom.js]: I think these should be renamed also to use `dropLeadingZeroes`.

**Comment by AusIV** [comment]
Sorry, I've been busy. I'l knock this out this morning.

**Comment by holgerd77** [comment]
:+1:

**Comment by AusIV** [comment]
It looks like the tests failed when trying to report coverage via coveralls. I'm not sure how to resolve this.

**Comment by holgerd77** [comment]
Hmm, not sure, we had some recent related changes. Will have a look tomorrow.

**Comment by holgerd77** [comment]
@AusIV thanks for updating this!

**Pull request merged**